File: src/main/java/com/gullygram/backend/service/SearchService.java
package com.gullygram.backend.service;

import com.gullygram.backend.dto.response.PostResponse;
import com.gullygram.backend.dto.response.UserSummary;
import com.gullygram.backend.entity.Post;
import com.gullygram.backend.entity.UserProfile;

import java.util.List;
import java.util.UUID;

public interface SearchService {
    List<UserSummary> searchUsers(String query, UUID currentUserId);
    List<PostResponse> searchPosts(String query, UUID currentUserId, Double lat, Double lon, Double radiusKm);
    List<PostResponse> searchHashtags(String hashtag, UUID currentUserId, Double lat, Double lon, Double radiusKm);
}

File: frontend/src/services/searchService.ts
import { api } from './api';
import { Post } from '../types/Post';
import { UserSummary } from '../types/User';

export const searchService = {
    searchUsers: async (query: string): Promise<UserSummary[]> => {
        const response = await api.get<UserSummary[]>('/search/users', {
            params: { q: query },
        });
        return response.data;
    },

    searchPosts: async (query: string, lat?: number, lon?: number, radiusKm?: number): Promise<Post[]> => {
        const response = await api.get<Post[]>('/search/posts', {
            params: { q: query, lat, lon, radiusKm },
        });
        return response.data;
    },

    searchHashtags: async (hashtag: string, lat?: number, lon?: number, radiusKm?: number): Promise<Post[]> => {
        const response = await api.get<Post[]>('/search/hashtags', {
            params: { q: hashtag, lat, lon, radiusKm },
        });
        return response.data;
    },
};
